--- tile.lua	2013-01-29 17:01:13.461898665 -0600
+++ uselessgap.lua	2013-01-30 12:59:41.794402805 -0600
@@ -63,11 +63,47 @@
         geom[height] = math.floor(unused * fact[i] / total_fact) - cls[c].border_width * 2
         geom[x] = group.coord
         geom[y] = coord
-        geom = cls[c]:geometry(geom)
         coord = coord + geom[height] + cls[c].border_width * 2
         unused = unused - geom[height] - cls[c].border_width * 2
         total_fact = total_fact - fact[i]
         used_size = math.max(used_size, geom[width]) + cls[c].border_width * 2
+
+        -- Useless gap.
+        local useless_gap = 25
+        if useless_gap > 0
+        then
+            -- Top and left clients are shrinked by two steps and
+            -- get moved away from the border. Other clients just
+            -- get shrinked in one direction.
+
+            top = false
+            left = false
+
+            if geom[y] == wa[y] then
+                top = true
+            end
+
+            if geom[x] == 0 or geom[x] == wa[x] then
+                left = true
+            end
+
+            if top then
+                geom[height] = geom[height] - 2 * useless_gap
+                geom[y] = geom[y] + useless_gap
+            else
+                geom[height] = geom[height] - useless_gap
+            end
+
+            if left then
+                geom[width] = geom[width] - 2 * useless_gap
+                geom[x] = geom[x] + useless_gap
+            else
+                geom[width] = geom[width] - useless_gap
+            end
+        end
+        -- End of useless gap.
+
+        geom = cls[c]:geometry(geom)
     end
 
     return used_size
